extras/kms/testing_test.go

```diff
@@ -16,3 +18,58 @@ func Test_stub(t *testing.T) {
 	t.Log("rows: ", rows)
 	require.NoError(t, err)
 }
+
+func Test_TestDeleteKeyPurpose_TestDeleteAllKeys(t *testing.T) {
+	const (
+		globalScope = "global"
+		orgScope    = "o_1234567890"
+	)
+	t.Parallel()
+	assert, require := assert.New(t), require.New(t)
+	ctx := context.Background()
+	db, _ := TestDb(t)
+	rw := dbw.New(db)
+	extWrapper := wrapping.NewTestWrapper([]byte(testDefaultWrapperSecret))
+
+	databaseKeyPurpose := KeyPurpose("database")
+
+	// init kms with a cache
+	kmsCache, err := New(rw, rw, []KeyPurpose{"database"})
+	require.NoError(err)
+	require.NoError(kmsCache.AddExternalWrapper(ctx, KeyPurposeRootKey, extWrapper))
+	// Make the global scope base keys
+	err = kmsCache.CreateKeys(ctx, globalScope, []KeyPurpose{databaseKeyPurpose})
+	require.NoError(err)
+
+	_, err = kmsCache.GetWrapper(ctx, globalScope, databaseKeyPurpose)
+	require.NoError(err)
+
+	TestDeleteKeyPurpose(t, db, databaseKeyPurpose)
```

Comment:
can this test be done in a way that confirms that this *only* deleted what it was supposed to, and did not just delete all of the keys?

Reply:
yep. ty!